python/lsst/afw/table/_base.py

```diff
@@ -115,6 +115,13 @@ def __getitem__(self, key):
         elif isinstance(key, np.ndarray):
             if key.dtype == bool:
                 return self.subset(key)
+            if np.issubdtype(key.dtype, np.integer) and key.ndim == 1:
+                # Return a shallow copy containing the desired records
+                new = type(self)(self.schema)
```

Comment:
I think you at least want to use `self.table` rather than `self.schema` here, since this is a shallow copy.  That should transfer shared metadata as well, though I'm a little worried that there might be subclasses for which `type(self)(...)` is not a reliable way to transfer all desired state to the new class.  I'm not sure there is - it's just something the base class can't guarantee, and that's the reason the other branches in this function call a method that derived classes can override.